src/index.js

```diff
@@ -80,27 +80,38 @@ function setup() {
 		'blockEditor.useSetting.before',
 		`wpcomvip-governance/nested-block-settings`,
 		( result, path, clientId, blockName ) => {
-			const hasCustomSetting =
-				// eslint-disable-next-line security/detect-object-injection
-				nestedSettingPaths[ blockName ] !== undefined &&
-				// eslint-disable-next-line security/detect-object-injection
-				nestedSettingPaths[ blockName ][ path ] === true;
-
-			if ( ! hasCustomSetting ) {
+			if ( ! blockName ) {
 				return result;
 			}
 
-			const blockNamePath = [
-				clientId,
-				...select( blockEditorStore ).getBlockParents( clientId, /* ascending */ true ),
-			]
-				.map( candidateId => select( blockEditorStore ).getBlockName( candidateId ) )
-				.reverse();
-
-			( { value: result } = getNestedSetting( blockNamePath, path, nestedSettings ) );
+			// iterate through the nestedSettingPaths to find the blockName
+			for ( const nestedBlockName in nestedSettingPaths ) {
```

Comment:
@alecgeatches - just need to pay special attention to this block as I touched it up a touch. I'd like to see if we could do some more pre-processing on the PHP side but that's for another PR.

Reply:
I've got one idea for optimization:

In JS prior to setting up this hook, we could split `nestedSettingPaths` into `nestedSettingWildcards` for names we need to evaluate like `core/*`, and a set of non-wildcards like `nestedSettingNonWildcards` (or a better name). Then we could keep the linear-time check that we had before using `hasCustomSetting` when matching non-wildcards, and only do a loop through any wildcards. If there are no wildcards, then there's no loop necessary.

Reply:
Overall I think this is pretty fast. I would split up wildcard and non-wildcard checks though.

Reply:
My thought was to try and do that on the PHP side so that we take advantage of server side rendering actually. That way we minimize the pre-processing necessary on the JS side 🤔